yt_dlp/extractor/nhk.py

```diff
@@ -334,3 +335,140 @@ def _real_extract(self, url):
             for x in traverse_obj(bangumi_list, ('part', ..., 'part-video-dasid')) or []]
 
         return self.playlist_result(bangumis, program_id, title, description)
+
+
+class NhkRadiruIE(InfoExtractor):
+    _GEO_COUNTRIES = ['JP']
+    IE_DESC = 'NHK らじる (Radiru/Rajiru)'
+    _VALID_URL = r'https://www\.nhk\.or\.jp/radio/(?:player/ondemand|ondemand/detail)\.html\?p=(?P<id>(?P<site>[\da-zA-Z]+)_(?P<corner>[\da-zA-Z]+)(?:_(?P<headline>[\da-zA-Z]+))?)'
+    # match https://www.nhk.or.jp/radio/player/ondemand.html (player) or https://www.nhk.or.jp/radio/ondemand/detail.html (programme page)
+    # then grab contents of p and split the numbers up into what they are in the api
+    _TESTS = [{
+        'url': 'https://www.nhk.or.jp/radio/player/ondemand.html?p=0449_01_3853544',
+        'skip': 'Episode expired on 2023-04-16',
+        '_type': 'video',
+        'info_dict': {
+            'channel': 'NHK-FM',
+            'description': '今回の前半は「ＮＥＷジャズ」特集と題して、曲名や演奏者の名前に「ＮＥＷ」がつく演奏や、新人の初リーダー作などを集めて聴いていく。',
+            'ext': 'm4a',
+            'id': '0449_01_3853544',
+            'series': 'ジャズ・トゥナイト',
+            'thumbnail': 'https://www.nhk.or.jp/prog/img/449/g449.jpg',
+            'timestamp': 1680969600,
+            'title': 'ジャズ・トゥナイト　ＮＥＷジャズ特集',
+            'upload_date': '20230408',
+            'release_timestamp': 1680962400,
+            'release_date': '20230408',
+            'was_live': True,
+        },
+    }, {
+        'url': 'https://www.nhk.or.jp/radio/ondemand/detail.html?p=0458_01',
+        '_type': 'playlist',
+        'info_dict': {
+            'id': '0458_01',
+            'title': 'ベストオブクラシック',
+            'description': '世界中の上質な演奏会をじっくり堪能する本格派クラシック番組。',
+            'channel': 'NHK-FM',
+            'thumbnail': 'https://www.nhk.or.jp/prog/img/458/g458.jpg',
+        },
+        'playlist_mincount': 3,  # airs every weekday so this should _hopefully_ be okay forever
+        'skip_download': True,
+    }, {
+        'url': 'https://www.nhk.or.jp/radio/player/ondemand.html?p=F300_06_3738470',  # one with letters in the id
+        '_type': 'video',
+        'skip': 'Expires on 2024-03-31',
+        'info_dict': {
+            'id': 'F300_06_3738470',
+            'ext': 'm4a',
+            'title': '有島武郎「一房のぶどう」',
+            'description': '朗読：川野一宇（ラジオ深夜便アンカー）\r\n\r\n（2016年12月8日放送「ラジオ深夜便『アンカー朗読シリーズ』」より）',
+            'channel': 'NHKラジオ第1、NHK-FM',
+            'timestamp': 1635757200,
+            'thumbnail': 'https://www.nhk.or.jp/radioondemand/json/F300/img/corner/box_109_thumbnail.jpg',
+            'release_date': '20161207',
+            'series': 'らじる文庫 by ラジオ深夜便 ',
+            'release_timestamp': 1481126700,
+            'upload_date': '20211101',
+        }
+    }
+    ]
+# https://www.nhk.or.jp/radionews/ known not working - doesnt use same api
+# https://www.nhk.or.jp/s-media/news/podcast/list/v1/all.xml podcast feed if you need it
```

Comment:
if they are not used by this extractor these comments are not pertinent imo
```suggestion
    }]
```

Reply:
the programme page isn't supported, but the actual player page is the same format as everything else
eg https://www.nhk.or.jp/radio/player/ondemand.html?p=F261_01_3855106
so it tries to extract it, and fails

Reply:
seems it uses [its own api](https://www.nhk.or.jp/s-media/news/news-site/list/v1/all.json)
still same format, just different url
could _maybe_ add a special case for site_id F261?
best way would be to figure out how it finds the api
[time passes]
never mind, they have a special case as well lol - see first bit of https://www.nhk.or.jp/radio/js/player_ondemand.js

Reply:
this would work for the F261 player page, yeah?
```diff
diff --git a/yt_dlp/extractor/nhk.py b/yt_dlp/extractor/nhk.py
index ce55fc6b8..0055be859 100644
--- a/yt_dlp/extractor/nhk.py
+++ b/yt_dlp/extractor/nhk.py
@@ -412,11 +412,14 @@ def _extract_episode_info(self, headline, programme_id, series_meta):
 
     def _real_extract(self, url):
         site_id, corner_id, headline_id = self._match_valid_url(url).group('site', 'corner', 'headline')
-
         programme_id = f'{site_id}_{corner_id}'
-        meta = self._download_json(
-            f'https://www.nhk.or.jp/radioondemand/json/{site_id}/bangumi_{programme_id}.json',
-            programme_id)['main']
+
+        if site_id == 'F261':
+            api_url = 'https://www.nhk.or.jp/s-media/news/news-site/list/v1/all.json'
+        else:
+            api_url = f'https://www.nhk.or.jp/radioondemand/json/{site_id}/bangumi_{programme_id}.json'
+
+        meta = self._download_json(api_url, programme_id)['main']
 
         series_meta = traverse_obj(meta, {
             'title': 'program_name',
```
.
> https://www.nhk.or.jp/radionews/ known not working - doesnt use same api
> the programme page isn't supported, but the actual player page is the same format as everything else

are you saying that this url:
https://www.nhk.or.jp/radionews/ 
should be handled as if it was this url?
https://www.nhk.or.jp/radio/player/ondemand.html?p=F261_01

if so, we could add an extractor class that matches only https://www.nhk.or.jp/radionews/ and returns a url_result of https://www.nhk.or.jp/radio/player/ondemand.html?p=F261_01

Reply:
added news support
not bothering with id F139 because it seems to be dev-only

Reply:
> this would work for the F261 player page, yeah?

sorry, beat you to it lol

Reply:
> are you saying that this url:
https://www.nhk.or.jp/radionews/
should be handled as if it was this url?
https://www.nhk.or.jp/radio/player/ondemand.html?p=F261_01

no, like https://www.nhk.or.jp/radio/ondemand/detail.html?p=F261_01
but yes, general idea is right

> if so, we could add an extractor class that matches only https://www.nhk.or.jp/radionews/ and returns a url_result of https://www.nhk.or.jp/radio/ondemand/detail.html?p=F261_01

yeah, that would probably be better than what i've actually done
uhhh

Reply:
> if so, we could add an extractor class that matches only https://www.nhk.or.jp/radionews/ and returns a url_result of https://www.nhk.or.jp/radio/ondemand/detail.html?p=F261_01

done